python/pyspark/sql/dataframe.py

```diff
@@ -1332,7 +1333,7 @@ def head(self, n=None):
         """
         if n is None:
             rs = self.head(1)
-            return rs[0] if rs else None
+            return rs[0] if rs else []
```

Comment:
Can you update the migration guide at https://github.com/apache/spark/blob/master/docs/pyspark-migration-guide.md? You can create a new title "Upgrading from PySpark 3.0 to 3.1"

Reply:
done